lib/agentx.c

```diff
@@ -23,25 +23,62 @@
 #include "libfrr.h"
 #include "xref.h"
 #include "lib/libagentx.h"
+#include "ringbuf.h"	 /* for ringbuf_remain, ringbuf_peek, ringbuf_.. */
+#include "frr_pthread.h" /* for struct frr_pthread */
+#include "bgpd/bgpd.h"	 /* for bgp_pth_ax */
 
 XREF_SETUP();
 
+
+extern int agentx_stop(struct frr_pthread *fpt, void **result);
+
 DEFINE_HOOK(agentx_enabled, (), ());
 
 //bool agentx_enabled = false;
 
 static struct event_loop *agentx_tm;
+static struct event_loop *main_tm;
 static struct event *timeout_thr = NULL;
 static struct list *events = NULL;
 
-static void agentx_events_update(void);
+struct frr_pthread *agentx_pth;
+
+/* buffer dedicated to "master -> agentx" threads */
+static struct ringbuf *ibuf_ax;
+#define RINGBUF_NB_TRAP 200
+/* mutex dedicated to trap transfert between threads */
+pthread_mutex_t ax_mtx;
+/* mutex dedicated to send/read exclusion */
+pthread_mutex_t ax_io_mtx;
+
+static void agentx_pthreads_init(void)
+{
+	assert(!agentx_pth);
+
+	struct frr_pthread_attr ax = {
+		.start = frr_pthread_attr_default.start,
+		.stop = frr_pthread_attr_default.stop,
+	};
+
+	agentx_pth = frr_pthread_new(&ax, "BGP Agentx thread", "bgpd_ax");
+}
```

Comment:
Why should it be BGP only into the library ?

Reply:
taken into account